### PR TITLE
Create users

### PR DESCRIPTION
**Pull request by panagiotismak**

### CODE REVIEW

File: app/controllers/users_controller.rb [hunk 1]
```diff
@@ -0,0 +1,21 @@
...
+  def new
+   @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      flash[:success] = "Welcome to Skroutz Lists #{@user.username}"
+      redirect_to root_path
+  	else
```
Comment (filippiazikou) [inline app/controllers/users_controller.rb]: Fix indentation here

File: app/controllers/users_controller.rb [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+      render 'new'
```
Comment (filippiazikou) [inline app/controllers/users_controller.rb]: Do you want to flash an error message here maybe?

File: app/models/user.rb [hunk 3]
```diff
@@ -0,0 +1,8 @@
+class User < ActiveRecord::Base
+  before_save { self.email = email.downcase }
```
Comment (filippiazikou) [inline app/models/user.rb]: empty line after `before_save`

File: app/models/user.rb [hunk 4]
```diff
@@ -0,0 +1,8 @@
...
+  validates :username, presence: true, uniqueness: { case_sensitive: false }, length: { minimum: 3, maximun: 25 }
+  VALID_EMAIL_REGEX = /\A[\w+\-.]+@[a-z\d\-.]+\.[a-z]+\z/i
+  validates :email, presence: true, uniqueness: { case_sensitive: false}, length: { maximum: 105 }, format: { with: VALID_EMAIL_REGEX }
+  has_secure_password
```
Comment (filippiazikou) [inline app/models/user.rb]: change lines for readability

**Review by filippiazikou — CHANGES_REQUESTED** [review]

File: app/models/user.rb (hunk 3 above)
Comment (panagiotismak) [reply app/models/user.rb]: done

File: app/models/user.rb (hunk 4 above)
Comment (panagiotismak) [reply app/models/user.rb]: done

File: app/controllers/users_controller.rb (hunk 1 above)
Comment (panagiotismak) [reply app/controllers/users_controller.rb]: not valid

**Comment by panagiotismak** [comment]
@filippiazikou take a look again

**Comment by filippiazikou** [comment]
Do you need all these images on git?

File: app/controllers/sessions_controller.rb [hunk 5]
```diff
@@ -0,0 +1,25 @@
...
+  end
+
+  def create
+    user = User.find_by(email: params[:session][:email].downcase)
+    if user && user.authenticate(params[:session][:password])
+  	  session[:user_id] =user.id
+      flash[:success] = "You have successfully logged in"
+      redirect_to user_path(user)
+    else
+  	  flash[:danger] = "There was something wrong with your login information. Please Try again"
```
Comment (filippiazikou) [inline app/controllers/sessions_controller.rb:14]: indentation

File: app/controllers/users_controller.rb [hunk 6]
```diff
@@ -0,0 +1,61 @@
...
+   @user = User.new
+  end
+
+  def create
+    @user = User.new(user_params)
+    if @user.save
+      session[:user_id] =@user.id
+      flash[:success] = "Welcome to Skroutz Lists #{@user.username}"
+      redirect_to user_path(@user)
+  	else
```
Comment (filippiazikou) [inline app/controllers/users_controller.rb:17]: indentation

**Review by filippiazikou — APPROVED** [review]
Congrats :D !!

File: app/controllers/sessions_controller.rb (hunk 5 above)
Comment (panagiotismak) [reply app/controllers/sessions_controller.rb:14]: done

File: app/controllers/users_controller.rb (hunk 6 above)
Comment (panagiotismak) [reply app/controllers/users_controller.rb:17]: done

**Pull request merged**